### PR TITLE
Abdul e2e

### PR DESCRIPTION
**Pull request by abdul-qadir**

### CODE REVIEW

File: account/login/login.data.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+
```
Comment (parthshah) [inline account/login/login.data.js]: This directory and files should be under tests/e2e not in the top level directory

File: gulpfile.js [hunk 2]
```diff
@@ -391,38 +410,19 @@ gulp.task('deploy', ['build'], function() {
   // create a new publisher
   var publisher = $.awspublish.create(awsConfig);
 
+  // define custom headers
+  var headers = {
+    'Cache-Control': 'max-age=94608000, no-transform, public'
+  };
+
   log('Deploying to S3');
```
Comment (parthshah) [inline gulpfile.js]: please merge these changes correctly. Why do we have gulpfile.js.orig

**Comment by parthshah** [comment]
Hi @abdul-qadir , I've added some comments let me know if you have any questions.
Also, please note that we have switched to a popup mode for login / register with social plugins. Here's an article that might help modify the test if necessary - https://ngmilk.rocks/2014/10/14/refactoring-auth0-login-tests-on-angularjs-with-protractor/

File: gulp.config.js [hunk 3]
```diff
@@ -6,6 +6,8 @@ module.exports = function() {
   var temp           = './.tmp/';
   var wiredep        = require('wiredep');
   var bowerFiles     = wiredep({devDependencies: true})['js'];
+  var e2e            ='./tests/e2e/';
+  var curEnv         = process.env.ENVIRONMENT || 'development';
```
Comment (vikasrohit) [inline gulp.config.js]: I guess, we can omit the use of environment variable in config file by using [gulp-filter](https://www.npmjs.com/package/gulp-filter) in the pipe after src operation in the task in gulpfile.js.

File: gulpfile.js [hunk 4]
```diff
@@ -359,10 +359,30 @@ gulp.task('serve-build', ['build'], function() {
...
+
+gulp.task('e2eCopy', function() {
+	  log('Copying environment specific files');
+	  
+	  return gulp
+	    .src(config.e2eTests)
+	    .pipe(gulp.dest(config.e2eTemp));
+	});
+
 gulp.task('e2e', [], function(done) {
```
Comment (vikasrohit) [inline gulpfile.js]: we can dependency of 'e2eDataFilesRename' and 'e2eCopy' tasks for e2e task so that we don't have to run them manually.

**Comment by vikasrohit** [comment]
@abdul-qadir we can add dependency of gulp-rename module of npm in package.json.

**Comment by nlitwin** [comment]
@parthshah Any thoughts on this one?

**Comment by nlitwin** [comment]
If we need the code from here, we can take it from the commits. But I think we can close this.

**Pull request closed without merging**